ndk/src/media_error.rs

```diff
@@ -62,10 +62,6 @@ pub enum MediaError {
     DrmLicenseExpired = ffi::media_status_t::AMEDIA_DRM_LICENSE_EXPIRED.0,
     #[doc(alias = "AMEDIA_IMGREADER_ERROR_BASE")]
     ImgreaderErrorBase = ffi::media_status_t::AMEDIA_IMGREADER_ERROR_BASE.0,
-    #[doc(alias = "AMEDIA_IMGREADER_NO_BUFFER_AVAILABLE")]
-    ImgreaderNoBufferAvailable = ffi::media_status_t::AMEDIA_IMGREADER_NO_BUFFER_AVAILABLE.0,
-    #[doc(alias = "AMEDIA_IMGREADER_MAX_IMAGES_ACQUIRED")]
-    ImgreaderMaxImagesAcquired = ffi::media_status_t::AMEDIA_IMGREADER_MAX_IMAGES_ACQUIRED.0,
```

Comment:
I'm quite certain (but please back me up on this) that only the 4 functions touched here can return this error code. I don't want users to go and try to match on these `MediaError` constants going forward if there are now special `AcquireResult` variants. If it turns out there are functions that do return this code (which gets turned into `__Unknown`), that's an error on our side that we should rectify.

For the lock errors below, I think they're more-or-less true errors (user tries to lock an image that's not been created for CPU access) and don't need a more obvious handling than these functions.

I'm a bit skeptical how that fares for the acquire functions. After all the first revision of this PR removed special-casing entirely, so that users looking for `NoBufferAvailable` had to match the `Err` themselves. And in a way they can "prevent this invalid(?) state": for example an `ImageReader` bound to an `EGL` `Context`/`Surface` will not return an `Image` (not even with a fence) until `SwapBuffers()` has been called.

Reply:
> I'm quite certain (but please back me up on this) that only the 4 functions touched here can return this error code.

yeah, that's also how it looks to me